From 67a590b6276f2b218a8a64425497ee885d622993 Mon Sep 17 00:00:00 2001
From: Alexandru Costache <alexandru@balena.io>
Date: Thu, 21 May 2020 16:36:03 +0200
Subject: [PATCH] Integrate machine independent resin environment configuration

This patch integrates resin default environment configuration in an u-boot which
has a version with Kconfig support.

This patch is a rebase of the original one in meta-balena
as that one no longer applies cleanly.

Signed-off-by: Andrei Gherzan <andrei@resin.io>
Signed-off-by: Alexandru Costache <alexandru@balena.io>

Upstream-Status: Inappropriate [configuration]
---
 include/env_default.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/include/env_default.h b/include/env_default.h
index 56a8bae39a..02f640d8bb 100644
--- a/include/env_default.h
+++ b/include/env_default.h
@@ -9,6 +9,8 @@
 
 #include <env_callback.h>
 
+#include <env_resin.h>
+
 #ifdef DEFAULT_ENV_INSTANCE_EMBEDDED
 env_t embedded_environment __UBOOT_ENV_SECTION__(environment) = {
 	ENV_CRC,	/* CRC Sum */
@@ -21,6 +23,7 @@ static char default_environment[] = {
 #else
 const uchar default_environment[] = {
 #endif
+	RESIN_ENV
 #ifndef CONFIG_USE_DEFAULT_ENV_FILE
 #ifdef	CONFIG_ENV_CALLBACK_LIST_DEFAULT
 	ENV_CALLBACK_VAR "=" CONFIG_ENV_CALLBACK_LIST_DEFAULT "\0"
-- 
2.17.1

